etna/transforms/decomposition/deseasonal.py

```diff
@@ -0,0 +1,211 @@
+        if df[self.in_column].isnull().values.any():
+            raise ValueError("The input column contains NaNs in the middle of the series! Try to use the imputer.")
+        self._seasonal = seasonal_decompose(
+            x=df[self.in_column], model=self.model, filt=None, two_sided=False, extrapolate_trend=0
+        ).seasonal[: self.period]
+        return self
+
+    def transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Subtract seasonal component.
+
+        Parameters
+        ----------
+        df:
+            Features dataframe with time
+
+        Returns
+        -------
+        result: pd.DataFrame
+            Dataframe with extracted features
+        """
+        result = df
+        seasonal = self._roll_seasonal(result[self.in_column])
+        if self.model == "additive":
+            result[self.in_column] -= seasonal
+        else:
+            if np.any(result[self.in_column] <= 0):
+                raise ValueError(
+                    "The input column contains zero or negative values,"
+                    "but multiplicative seasonality can not work with such values."
+                )
+            result[self.in_column] /= seasonal
+        return result
+
+    def inverse_transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Add seasonal component.
+
+        Parameters
+        ----------
+        df:
+            Features dataframe with time
+
+        Returns
+        -------
+        result: pd.DataFrame
+            Dataframe with extracted features
+        """
+        result = df
+        seasonal = self._roll_seasonal(result[self.in_column])
+        if self.model == "additive":
+            result[self.in_column] += seasonal
+        else:
+            if np.any(result[self.in_column] <= 0):
+                raise ValueError(
+                    "The input column contains zero or negative values,"
+                    "but multiplicative seasonality can not work with such values."
+                )
+            result[self.in_column] *= seasonal
+        if self.in_column == "target":
+            quantiles = match_target_quantiles(set(result.columns))
+            for quantile_column_nm in quantiles:
+                if self.model == "additive":
+                    result.loc[:, quantile_column_nm] += seasonal
+                else:
+                    if np.any(result.loc[quantile_column_nm] <= 0):
+                        raise ValueError(
+                            f"The {quantile_column_nm} column contains zero or negative values,"
+                            "but multiplicative seasonality can not work with such values."
+                        )
+                    result.loc[:, quantile_column_nm] *= seasonal
+        return result
+
+
+class DeseasonalityTransform(ReversiblePerSegmentWrapper):
+    """Transform that uses :py:class:`statsmodels.tsa.seasonal.seasonal_decompose` to subtract season from the data.
```

Comment:
Is `seasonal_decompose` a class? You should probably write: `py:func:`.

Reply:
May be it is better to write: `to subtract seasonal component from the data` or `to subtract seasonal pattern from the data` or smth like this.